jupyter-js-widgets/src/manager-base.js

```diff
@@ -426,11 +432,15 @@ ManagerBase.prototype.set_state = function(state, displayOptions) {
         return Promise.all(_.map(models, function(model) {
             // Display the views of the model.
             if (state[model.id] !== undefined) {
-                return Promise.all(_.map(state[model.id].views, function(options) {
-                    // Display the model using the display options merged with the
-                    // options.
-                    return that.display_model(undefined, model, Object.assign({}, options, displayOptions));
-                }));
+                // Display the model using the display options merged with the
+                // options.
+                if (displayOptions.displayOnce && state[model.id].views) {
```

Comment:
Changes the default set_state behavior from displaying a widget view for every "view" in the state to only once.